Help request, trying to batch write with writeprecision::MS

I use influxdbV1 with batch writing, Writepresision::MS (or NS/US for that matter)
influxdb client library V3.2
ESP8266 board 2.7.1

And a HX711 with the latest library on pins
const int LOADCELL_DOUT_PIN = 13;
const int LOADCELL_SCK_PIN = 12;

My output in Grafana looks like this, data gets swapped


2020-06-14 18:52:50 0
2020-06-14 18:52:50 0
2020-06-14 18:52:50 0
2020-06-14 18:52:49 500
2020-06-14 18:52:49 499
2020-06-14 18:52:49 500
2020-06-14 18:52:49 500
2020-06-14 18:52:49 0
2020-06-14 18:52:49 12
2020-06-14 18:52:49 499
2020-06-14 18:52:49 500
2020-06-14 18:52:49 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 499
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:48 500
2020-06-14 18:52:47 245
2020-06-14 18:52:47 162
2020-06-14 18:52:47 0
2020-06-14 18:52:47 0
2020-06-14 18:52:47 500
2020-06-14 18:52:47 476
2020-06-14 18:52:47 464
2020-06-14 18:52:47 449
2020-06-14 18:52:47 379
2020-06-14 18:52:46 0
2020-06-14 18:52:46 0


The output from println is okay so I'm trying to write in the right order

18:52:46.867 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153566
18:52:46.970 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153567
18:52:47.176 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153567
18:52:47.278 -> Data written to InfluxDB :-) Massa: 162.00 Tijd: 1592153567
18:52:47.381 -> Data written to InfluxDB :-) Massa: 245.00 Tijd: 1592153567
18:52:47.483 -> Data written to InfluxDB :-) Massa: 379.00 Tijd: 1592153567
18:52:47.586 -> Data written to InfluxDB :-) Massa: 449.00 Tijd: 1592153567
18:52:47.688 -> Data written to InfluxDB :-) Massa: 464.00 Tijd: 1592153567
18:52:47.791 -> Data written to InfluxDB :-) Massa: 476.00 Tijd: 1592153567
18:52:47.895 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153567
18:52:47.998 -> Data written to InfluxDB :-) Massa: 499.00 Tijd: 1592153568
18:52:48.100 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.204 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.306 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.408 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.512 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.615 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.718 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.821 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.926 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153568
18:52:48.994 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153569
18:52:49.096 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153569
18:52:49.269 -> Data written to InfluxDB :-) Massa: 499.00 Tijd: 1592153569
18:52:49.373 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153569
18:52:49.476 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153569
18:52:49.578 -> Data written to InfluxDB :-) Massa: 500.00 Tijd: 1592153569
18:52:49.680 -> Data written to InfluxDB :-) Massa: 499.00 Tijd: 1592153569
18:52:49.784 -> Data written to InfluxDB :-) Massa: 12.00 Tijd: 1592153569
18:52:49.886 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153569
18:52:49.987 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153570
18:52:50.091 -> Data written to InfluxDB :-) Massa: 0.00 Tijd: 1592153570


If I try to manually set the time to resolve this (dPoint.setTime( tnow )) or dPoint.setTime( tijd )
the following happens:

InfluxDB write failed: {"error":"partial write: points beyond retention policy dropped=20"}

It even goes wrong when I set
#define MAX_BATCH_SIZE 1
#define WRITE_BUFFER_SIZE 1
effectively disabling batch-write :-S

What am I doing wrong?



My code is straightforward:

`/**
 * Secure Write Example code for InfluxDBClient library for Arduino
 * Enter WiFi and InfluxDB parameters below
 *
 * Demonstrates connection to any InfluxDB instance accesible via:
 *  - unsecured http://...
 *  - secure https://... (appropriate certificate is required)
 *  - InfluxDB 2 Cloud at https://cloud2.influxdata.com/ (certificate is preconfigured)
 * Measures signal level of all visible WiFi networks including signal level of the actually connected one
 * This example demonstrates time handling, how to write measures with different priorities, batching and retry
 * Data can be immediately seen in a InfluxDB 2 Cloud UI - measurements wifi_status and wifi_networks
 **/

#if defined(ESP32)
#include <WiFiMulti.h>
WiFiMulti wifiMulti;
#define DEVICE "ESP32"
#elif defined(ESP8266)
#include <ESP8266WiFiMulti.h>
ESP8266WiFiMulti wifiMulti;
#define DEVICE "ESP8266"
#define WIFI_AUTH_OPEN ENC_TYPE_NONE
#endif

#include <InfluxDbClient.h>
#include <InfluxDbCloud.h>


// Setup the loadcell stuff
#include <HX711.h>  // HX711 library for the scale

#define calibration_factor 923.53 //-7050.0 //This value is obtained using the SparkFun_HX711_Calibration sketch tbv china loadcell
// #define calibration_factor 701.77 //-7050.0 //This value is obtained using the SparkFun_HX711_Calibration sketch tbv dual 3kg tedea
#define calibration_factor2 3694.12 //-7050.0 //This value is obtained using the SparkFun_HX711_Calibration sketch tbv china loadcell


const int LOADCELL_DOUT_PIN = 13;
const int LOADCELL_SCK_PIN = 12;

HX711 scale;

// weight values
float weight = 0.0;   // measured weight
int weightout = 0;    // integer weight out
float dyntar = 0.0;   // dynamic tar
const int freq = 80;  // HX711 supports 10 and 80 Hertz, default = 10, hardware mod necessary
uint64_t tijd = 0;     // test with usec unixtime, didn't work
int t = 0;            // save measurement interval millis incrementals
#define THRESHOLD 2.1 // Measures only if the weight is greater than 2 gram.


// WiFi AP SSID
#define WIFI_SSID  ********
// WiFi password
#define WIFI_PASSWORD  ********



// Influxdb1 @home
#define INFLUXDB_URL  ********
#define INFLUXDB_DB_NAME ********
#define INFLUXDB_USER  ********
#define INFLUXDB_PASSWORD  ********

// #define INFLUXDB_BUCKET "Weegschaal"

// Influxdb2 connect
// InfluxDB client instance with preconfigured InfluxCloud certificate
// InfluxDBClient client(INFLUXDB_URL, INFLUXDB_ORG, INFLUXDB_BUCKET, INFLUXDB_TOKEN, InfluxDbCloud2CACert);

// Influxdb1 connect
// InfluxDB client instance for InfluxDB 1
InfluxDBClient client(INFLUXDB_URL, INFLUXDB_DB_NAME);
// Zie ook in de setup loop!

// Set timezone string according to https://www.gnu.org/software/libc/manual/html_node/TZ-Variable.html
// Examples:
//  Pacific Time:   "PST8PDT"
//  Eastern:        "EST5EDT"
//  Japanesse:      "JST-9"
//  Central Europe: "CET-1CEST,M3.5.0,M10.5.0/3"
//#define TZ_INFO "CET-1CEST,M3.5.0,M10.5.0/3"
#define TZ_INFO "CET-1CEST-2,M3.5.0/2,M10.5.0/3"

#define WRITE_PRECISION WritePrecision::MS
#define MAX_BATCH_SIZE 10
#define WRITE_BUFFER_SIZE 30
#define MAX_TIME 60
#define HTTP_KEEP true

// Data point
Point sensorStatus("wifi_status");

// Number for loops to sync time using NTP
int iterations = 0;

void timeSync() {
  // Synchronize UTC time with NTP servers
  // Accurate time is necessary for certificate validaton and writing in batches
  configTime(0, 0, "0.nl.pool.ntp.org", "3.nl.pool.ntp.org", "nl.pool.ntp.org" );
  // Set timezone
  setenv("TZ", TZ_INFO, 1);

  // Wait till time is synced
  Serial.print("Syncing time");
  int i = 0;
  while (time(nullptr) < 1000000000ul && i < 1000) {
    Serial.print(".");
    delay(100);
    i++;
  }
  Serial.println();

  // Show time
  time_t tnow = time(nullptr);
  Serial.print("Synchronized time: ");
  Serial.println(String(ctime(&tnow)));
}

void setup() {
  Serial.begin(115200);

  // Setup the scale
  scale.begin(LOADCELL_DOUT_PIN, LOADCELL_SCK_PIN);
  scale.set_scale(calibration_factor); //This value is obtained by using the SparkFun_HX711_Calibration sketch
  delay(1000);
  scale.tare(); //Assuming there is no weight on the scale at start up, reset the scale to 0


  // Setup wifi
  WiFi.mode(WIFI_STA);
  wifiMulti.addAP(WIFI_SSID, WIFI_PASSWORD);

  Serial.print("Connecting to wifi");
  while (wifiMulti.run() != WL_CONNECTED) {
    Serial.print(".");
    delay(100);
  }
  Serial.println();

  // Add tags
//  sensorStatus.addTag("device", DEVICE);
//  sensorStatus.addTag("SSID", WiFi.SSID());

  // Sync time for certificate validation
  timeSync();

  // Check server connection
  if (client.validateConnection()) {
    Serial.print("Connected to InfluxDB: ");
    Serial.println(client.getServerUrl());
  } else {
    Serial.print("InfluxDB connection failed: ");
    Serial.println(client.getLastErrorMessage());
  }

  // Set InfluxDB 1 authentication params
  client.setConnectionParamsV1(INFLUXDB_URL, INFLUXDB_DB_NAME, INFLUXDB_USER, INFLUXDB_PASSWORD);

  //Enable messages batching and retry buffer
  client.setWriteOptions(WRITE_PRECISION, MAX_BATCH_SIZE, WRITE_BUFFER_SIZE, MAX_TIME, HTTP_KEEP);
}

void loop() {
  // Sync time for batching once per hour
  if ( (millis()/1000/3600) > iterations ) {
  timeSync();
  iterations++;
  }

  if (millis() > t + (1000/freq*2)) {  
    weight = (scale.get_units());
    // dynamic tar if below threshold
    if ( (weight + dyntar) < THRESHOLD ) { weight = -dyntar; }
    weightout = int (weight + dyntar);
    writeData(weightout);
    t = millis();
  }
  delay(100);
}

void writeData(float weight) {  

    time_t tnow = time(nullptr);

    tijd = xx_time_get_time();

    Point dPoint("Massa");  
    dPoint.addField("gram", weight);  
//    dPoint.setTime( tijd );  //set the time
//    dPoint.setTime( tnow );  //set the time
    Serial.print("Massa: ");  
    Serial.print(weight);  
    Serial.print(" Tijd: ");
    Serial.println(tnow);
//    Serial.print(" Tijd2: ");
//    print((long)tijd);
    if(!client.writePoint(dPoint)) {
        Serial.print("InfluxDB write failed: ");
        Serial.println(client.getLastErrorMessage());
    }   else {
      Serial.print("Data written to InfluxDB :-) ");
    }
}

/**
#include <math.h>
void print(uint64_t value)
{
    const int NUM_DIGITS    = log10(value) + 1;
    char sz[NUM_DIGITS + 1];
    sz[NUM_DIGITS] =  0;
    for ( size_t i = NUM_DIGITS; i--; value /= 10)
    {
        sz[i] = '0' + (value % 10);
    }
     Serial.println(sz);
}
**/

int64_t xx_time_get_time() {
  struct timeval tv;
  gettimeofday(&tv, NULL);
  return (tv.tv_sec * 1000LL + (tv.tv_usec / 1000LL));
}`

as soon as I enable debugging it doesn't compile, even in a fresh portable setup with just the libraries I need and on a generic esp8266 or sparkfun esp8266 thing board

Arduino: 1.8.12 (Windows 10), Board: "SparkFun ESP8266 Thing, 80 MHz, Flash, Legacy (new can return nullptr), All SSL ciphers (most compatible), 512KB (FS:32KB OTA:~230KB), v2 Lower Memory, Disabled, None, Only Sketch, 115200"

In file included from C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/stdio.h:6:0,

                 from C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/stdio.h:63,

                 from C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1\cores\esp8266/Arduino.h:32,

                 from C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.h:32,

                 from C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:28:

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp: In member function 'bool InfluxDBClient::init()':

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:199:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("Init\n"));

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:199:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("Init\n"));

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:200:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Server url: %s\n"), _serverUrl.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:200:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Server url: %s\n"), _serverUrl.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:201:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Org: %s\n"), _org.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:201:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Org: %s\n"), _org.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:202:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Bucket: %s\n"), _bucket.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:202:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Bucket: %s\n"), _bucket.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:203:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Token: %s\n"), _authToken.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:203:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  Token: %s\n"), _authToken.c_str());

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:204:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  DB version: %d\n"), _dbVersion);

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:204:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("  DB version: %d\n"), _dbVersion);

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp: In member function 'void InfluxDBClient::setUrls()':

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:274:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("setUrls\n"));

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:274:5: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

     INFLUXDB_CLIENT_DEBUG(F("setUrls\n"));

     ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:281:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:281:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:285:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  queryUrl: %s\n"), _queryUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:285:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  queryUrl: %s\n"), _queryUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:300:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:300:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:301:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  queryUrl: %s\n"), _queryUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:301:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  queryUrl: %s\n"), _queryUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: initializer fails to determine size of '__c'

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:306:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\packages\esp8266\hardware\esp8266\2.7.1/tools/sdk/libc/xtensa-lx106-elf/include/sys/pgmspace.h:39:236: error: array must be initialized with a brace-enclosed initializer

     #define PSTR(s) (__extension__({static const char __c[] __attribute__((__aligned__(4))) __attribute__((section( "\".irom0.pstr." __FILE__ "." __STRINGIZE(__LINE__) "."  __STRINGIZE(__COUNTER__) "\", \"aSM\", @progbits, 1 #"))) = (s); &__c[0];}))

                                                                                                                                                                                                                                            ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\util/debug.h:34:66: note: in expansion of macro 'PSTR'

 # define INFLUXDB_CLIENT_DEBUG(fmt, ...) Serial.printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__ )

                                                                  ^

C:\Arduinolibs\arduino-1.8.12-windows\arduino-1.8.12\portable\sketchbook\libraries\InfluxDB-Client-for-Arduino-master\src\InfluxDbClient.cpp:306:9: note: in expansion of macro 'INFLUXDB_CLIENT_DEBUG'

         INFLUXDB_CLIENT_DEBUG(F("  writeUrl: %s\n"), _writeUrl.c_str());

         ^

exit status 1
Error compiling for board SparkFun ESP8266 Thing.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,8 @@
 # Changelog
+
+## Version 3.3.0 (in progress)
+- [FIX] More precice default timestamp generating, up to microseconds
+
 ## Version 3.2.0 (2020-06-09)
 - [NEW] Added possibility to read data from InfluxDB using Flux queries
 - [NEW] `timeSync` utility function for synchronous time synchronization using NTP 

--- a/README.md
+++ b/README.md
@@ -153,7 +153,7 @@ InfluxDB client for Arduino can write data in batches. A batch is simply a set o
 If using batch writes, the timestamp should be employed. Timestamp specifies the time where data was gathered and it is used in the form of a number of seconds (milliseconds, etc) from epoch (1.1.1970) UTC.
 If points have no timestamp assigned, InfluxDB assigns timestamp at the time of writing, which could happen much later than the data has been obtained, because final batch write will happen when the batch is full (or when [flush buffer](#buffer-handling-and-retrying) is forced).
 
-InfuxDB allows sending timestamp in various precisions - nanoseconds, microseconds, milliseconds or seconds. The milliseconds precision is usually enough for using on Arduino.
+InfuxDB allows sending timestamp in various precisions - nanoseconds, microseconds, milliseconds or seconds. The milliseconds precision is usually enough for using on Arduino. Maximum avavailable precision is microseconds. Setting to nanosecond will just add zeroes for microseconds fraction.
 
 The client has to be configured with time precision. The default settings is not using the timestamp. The `setWriteOptions` functions allow setting various parameters and one of them is __write precision__:
 ``` cpp
@@ -164,8 +164,10 @@ When a write precision is configured, the client will automatically assign curre
 
 If you want to manage timestamp on your own, there are several ways how to set timestamp explicitly.
 - `setTime(WritePrecision writePrecision)` - Sets timestamp to actual time in desired precision
-- `setTime(unsigned long seconds)` -  Sets timestamp in seconds since epoch. Write precision must be set to `S` 
-- `setTime(String timestamp)` - Set custom timestamp in precision specified in InfluxDBClient. 
+- `setTime(unsigned long long timestamp)` -  Sets timestamp in an offset since epoch. Correct precision must be set InfluxDBClient::setWriteOptions.
+- `setTime(String timestamp)` - Sets timestamp in an offset since epoch. Correct precision must be set InfluxDBClient::setWriteOptions.
+
+The `getTime()` method allows copying timestamp between points.
 
 
 ### Configure Time

--- a/src/InfluxDbClient.cpp
+++ b/src/InfluxDbClient.cpp
@@ -115,32 +115,30 @@ String Point::toLineProtocol() const {
 }
 
 void  Point::setTime(WritePrecision precision) {
-    static char buff[10];
-    time_t now = time(nullptr);
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    
     switch(precision) {
         case WritePrecision::NS:
-             sprintf(buff, "%06d000",  micros()%1000000uL);
-            _timestamp = String(now) + buff;
+            setTime(getTimeStamp(&tv,9));
             break;
         case WritePrecision::US:
-             sprintf(buff, "%06d",  micros()%1000000uL);
-            _timestamp = String(now) + buff;
+            setTime(getTimeStamp(&tv,6));
             break;
-        case WritePrecision::MS:
-             sprintf(buff, "%03d",  millis()%1000u);
-            _timestamp = String(now) + buff;
+        case WritePrecision::MS: 
+            setTime(getTimeStamp(&tv,3));
+            break;
+        case WritePrecision::S:
+            setTime(getTimeStamp(&tv,0));
             break;
         case WritePrecision::NoTime:
             _timestamp = "";
             break;
-        case WritePrecision::S:
-             _timestamp = String(now);
-             break;
     }
 }
 
-void  Point::setTime(unsigned long timestamp) {
-    _timestamp = String(timestamp);
+void  Point::setTime(unsigned long long timestamp) {
+    _timestamp = timeStampToString(timestamp);
 }
 
 void  Point::setTime(String timestamp) {

--- a/src/InfluxDbClient.h
+++ b/src/InfluxDbClient.h
@@ -83,9 +83,9 @@ class Point {
     void addField(String name, const char *value);
     // Set timestamp to `now()` and store it in specified precision, nanoseconds by default. Date and time must be already set. See `configTime` in the device API
     void setTime(WritePrecision writePrecision = WritePrecision::NS);
-    // Set timestamp in seconds since epoch (1.1.1970). Precision should be set to `S` 
-    void setTime(unsigned long seconds);
-    // Set timestamp in desired precision (specified in InfluxDBClient) since epoch (1.1.1970 00:00:00)
+    // Set timestamp in offset since epoch (1.1.1970). Correct precision must be set InfluxDBClient::setWriteOptions.
+    void setTime(unsigned long long timestamp);
+    // Set timestamp in offset since epoch (1.1.1970 00:00:00). Correct precision must be set InfluxDBClient::setWriteOptions.
     void setTime(String timestamp);
     // Clear all fields. Usefull for reusing point  
     void clearFields();
@@ -95,10 +95,12 @@ class Point {
     bool hasFields() const { return _fields.length() > 0; }
     // True if a point contains at least one tag
     bool hasTags() const   { return _tags.length() > 0; }
-     // True if a point contains timestamp
+    // True if a point contains timestamp
     bool hasTime() const   { return _timestamp.length() > 0; }
     // Creates line protocol
     String toLineProtocol() const;
+    // returns current timestamp
+    String getTime() const { return _timestamp; } 
   protected:
     String _measurement;
     String _tags;

--- a/src/util/helpers.cpp
+++ b/src/util/helpers.cpp
@@ -46,3 +46,30 @@ void timeSync(const char *tzInfo, const char* ntpServer1, const char* ntpServer2
   Serial.print("Synchronized time: ");
   Serial.println(ctime(&tnow));
 }
+
+unsigned long long getTimeStamp(struct timeval *tv, int secFracDigits) {
+    unsigned long long tsVal = 0;
+    switch(secFracDigits) {
+        case 0: 
+            tsVal = tv->tv_sec;
+            break;
+        case 6: 
+            tsVal = tv->tv_sec * 1000000LL + tv->tv_usec;
+            break;
+        case 9:
+            tsVal = tv->tv_sec * 1000000000LL + tv->tv_usec * 1000LL;
+            break;
+        case 3: 
+        default:
+            tsVal = tv->tv_sec * 1000LL + tv->tv_usec / 1000LL;
+            break;
+
+    }
+    return tsVal;
+}
+
+String timeStampToString(unsigned long long timestamp) {
+    static char buff[50];
+    snprintf(buff, 50, "%llu", timestamp);
+    return String(buff);
+}

--- a/src/util/helpers.h
+++ b/src/util/helpers.h
@@ -28,10 +28,17 @@
 #define _INFLUXDB_CLIENT_HELPERS_H
 
 #include <Arduino.h>
+#include <sys/time.h>  
 
 // Synchronize time with NTP servers and waits for completition. Prints waiting progress and final synchronized time to the serial.
 // Accurate time is necessary for certificate validion and writing points in batch
 // For the fastest time sync find NTP servers in your area: https://www.pool.ntp.org/zone/
 void timeSync(const char *tzInfo, const char* ntpServer1, const char* ntpServer2 = nullptr, const char* ntpServer3 = nullptr);
+
+// Create timestamp in offset from epoch. secFracDigits specify resulution. 0 - seconds, 3 - milliseconds, etc. Maximum and default is 9 - nanoseconds.
+unsigned long long getTimeStamp(struct timeval *tv, int secFracDigits = 3);
+
+// Converts unsigned long long timestamp to String
+String timeStampToString(unsigned long long timestamp);
 
 #endif //_INFLUXDB_CLIENT_HELPERS_H

--- a/test/TestSupport.h
+++ b/test/TestSupport.h
@@ -1,10 +1,14 @@
 #ifndef _TEST_SUPPORT_H_
 #define _TEST_SUPPORT_H_
 
-#define TEST_INIT(name) int temp = failures; const char *testName = name; do { Serial.println(testName)
-#define TEST_END()  } while(0); Serial.printf("%s %s\n",testName,failures == temp?"SUCCEEDED":"FAILED")
-#define TEST_ASSERT(a) if(testAssert(__LINE__, (a))) break
-#define TEST_ASSERTM(a,m) if(testAssertm(__LINE__, (a),(m))) break
+#define TEST_INIT(name)  int temp = failures;\
+ const char *testName = name; \
+  do { \
+  Serial.println(testName)
+#define TEST_END()  } while(0); \
+ end: Serial.printf("%s %s\n",testName,failures == temp?"SUCCEEDED":"FAILED")
+#define TEST_ASSERT(a) if(testAssert(__LINE__, (a))) goto end
+#define TEST_ASSERTM(a,m) if(testAssertm(__LINE__, (a),(m))) goto end
 
 #include "query/FluxParser.h"
 

--- a/test/test.ino
+++ b/test/test.ino
@@ -48,11 +48,10 @@ void setup() {
 }
 
 void loop() {
-    //tests
+    // tests
     testPoint();
     testFluxTypes();
     testFluxParserEmpty();
-    
     testFluxParserSingleTable();
     testFluxParserNilValue();
     testFluxParserMultiTables(false);
@@ -68,11 +67,11 @@ void loop() {
     testUserAgent();
     testFailedWrites();
     testTimestamp();
-    // testRetryOnFailedConnection();
-    // testBufferOverwriteBatchsize1();
-    // testBufferOverwriteBatchsize5();
-    // testServerTempDownBatchsize5();
-    // testRetriesOnServerOverload();
+    testRetryOnFailedConnection();
+    testBufferOverwriteBatchsize1();
+    testBufferOverwriteBatchsize5();
+    testServerTempDownBatchsize5();
+    testRetriesOnServerOverload();
 
     Serial.printf("Test %s\n", failures ? "FAILED" : "SUCCEEDED");
     while(1) delay(1000);
@@ -115,15 +114,22 @@ void testPoint() {
 
     TEST_ASSERT(!p.hasTime());
     time_t now = time(nullptr);
+    String snow(now);
     p.setTime(now);
-    String testLineTime = testLine + " " + now;
+    String testLineTime = testLine + " " + snow;
+    line = p.toLineProtocol();
+    TEST_ASSERTM(line == testLineTime, line);
+    
+    unsigned long long ts = now*1000000000LL+123456789;
+    p.setTime(ts);
+    testLineTime = testLine + " " + snow + "123456789";
     line = p.toLineProtocol();
     TEST_ASSERTM(line == testLineTime, line);
 
     now += 10;
-    String nowStr(now);
-    p.setTime(nowStr);
-    testLineTime = testLine + " " + nowStr;
+    snow = now;
+    p.setTime(snow);
+    testLineTime = testLine + " " + snow;
     line = p.toLineProtocol();
     TEST_ASSERTM(line == testLineTime, line);
 
@@ -132,7 +138,7 @@ void testPoint() {
     int partsCount;
     String *parts = getParts(line, ' ', partsCount);
     TEST_ASSERTM(partsCount == 3, String("3 != ") + partsCount);
-    TEST_ASSERT(parts[2].length() == nowStr.length());
+    TEST_ASSERT(parts[2].length() == snow.length());
     delete[] parts;
 
     p.setTime(WritePrecision::MS);
@@ -140,21 +146,21 @@ void testPoint() {
     line = p.toLineProtocol();
     parts = getParts(line, ' ', partsCount);
     TEST_ASSERT(partsCount == 3);
-    TEST_ASSERT(parts[2].length() == nowStr.length() + 3);
+    TEST_ASSERT(parts[2].length() == snow.length() + 3);
     delete[] parts;
 
     p.setTime(WritePrecision::US);
     line = p.toLineProtocol();
     parts = getParts(line, ' ', partsCount);
     TEST_ASSERT(partsCount == 3);
-    TEST_ASSERT(parts[2].length() == nowStr.length() + 6);
+    TEST_ASSERT(parts[2].length() == snow.length() + 6);
     delete[] parts;
 
     p.setTime(WritePrecision::NS);
     line = p.toLineProtocol();
     parts = getParts(line, ' ', partsCount);
     TEST_ASSERT(partsCount == 3);
-    TEST_ASSERT(parts[2].length() == nowStr.length() + 9);
+    TEST_ASSERT(parts[2].length() == snow.length() + 9);
     delete[] parts;
 
     p.clearFields();
@@ -719,6 +725,31 @@ void testFailedWrites() {
 
 void testTimestamp() {
     TEST_INIT("testTimestamp");
+
+    struct timeval tv;
+    tv.tv_usec = 1234;
+    tv.tv_sec = 5678;
+    unsigned long long ts = getTimeStamp(&tv, 0);
+    TEST_ASSERTM( ts == 5678, timeStampToString(ts));
+    ts = getTimeStamp(&tv, 3);
+    TEST_ASSERTM( ts == 5678001, timeStampToString(ts));
+    ts = getTimeStamp(&tv, 6);
+    TEST_ASSERTM( ts == 5678001234, timeStampToString(ts));
+    ts = getTimeStamp(&tv, 9);
+    TEST_ASSERTM( ts == 5678001234000, timeStampToString(ts));
+
+    // Test increasing timestamp
+    String prev = "";
+    for(int i = 0;i<2000;i++) {
+        Point p("test");
+        p.setTime(WritePrecision::US);
+        String act = p.getTime();
+        TEST_ASSERTM( i == 0 || prev < act, String(i) + ": " + prev + " vs " + act);
+        prev = act;
+        delayMicroseconds(100);
+    }
+
+
     serverLog(INFLUXDB_CLIENT_TESTING_URL, "testTimestamp");
     InfluxDBClient client(INFLUXDB_CLIENT_TESTING_URL, INFLUXDB_CLIENT_TESTING_ORG, INFLUXDB_CLIENT_TESTING_BUC, INFLUXDB_CLIENT_TESTING_TOK);
     client.setWriteOptions(WritePrecision::S, 1, 5);
@@ -977,6 +1008,7 @@ bool testFluxDateTimeValue(FluxQueryResult flux, int columnIndex,  const char *c
         TEST_ASSERTM(dt.microseconds == us,  String(dt.microseconds) + " vs " + String(us));
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -987,6 +1019,7 @@ bool testStringValue(FluxQueryResult flux, int columnIndex,  const char *columnN
         TEST_ASSERTM(flux.getValueByName(columnName).getRawValue() == rawValue, flux.getValueByName(columnName).getRawValue());
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -997,12 +1030,12 @@ bool testStringVector(std::vector<String> vect, const char *values[], int size) 
             if(vect[i] != values[i]) {
                 Serial.print("assert failure: ");
                 Serial.println(vect[i]);
-                goto fail;
+                goto end;
             }
         }
         return true;
     } while(0);
-fail:
+end:
     return false;
 }
 
@@ -1013,6 +1046,7 @@ bool testDoubleValue(FluxQueryResult flux, int columnIndex,  const char *columnN
         TEST_ASSERTM(flux.getValueByName(columnName).getRawValue() == rawValue, flux.getValueByName(columnName).getRawValue());
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -1023,6 +1057,7 @@ bool testLongValue(FluxQueryResult flux, int columnIndex,  const char *columnNam
         TEST_ASSERTM(flux.getValueByName(columnName).getRawValue() == rawValue, flux.getValueByName(columnName).getRawValue());
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -1033,6 +1068,7 @@ bool testUnsignedLongValue(FluxQueryResult flux, int columnIndex,  const char *c
         TEST_ASSERTM(flux.getValueByName(columnName).getRawValue() == rawValue, flux.getValueByName(columnName).getRawValue());
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -1046,6 +1082,7 @@ bool testTableColumns(FluxQueryResult flux,  const char *columns[], int columnsC
         TEST_ASSERTM(flux.getColumnIndex("x") == -1, "flux.getColumnIndex(\"x\")");
         return true;
     } while(0);
+end:
     return false;
 }
 
@@ -1460,7 +1497,7 @@ void initInet() {
     } else {
         Serial.printf("Connected to: %s (%d)\n", WiFi.SSID().c_str(), WiFi.RSSI());
 
-        timeSync("CET-1CEST,M3.5.0,M10.5.0/3", "pool.ntp.org", "0.cz.pool.ntp.org", "1.cz.pool.ntp.org");
+        timeSync("CET-1CEST,M3.5.0,M10.5.0/3", "0.cz.pool.ntp.org", "1.cz.pool.ntp.org", "pool.ntp.org");
 
         deleteAll(INFLUXDB_CLIENT_TESTING_URL);
     }

